lib/neuron/connection.ex

```diff
@@ -1,16 +1,7 @@
 defmodule Neuron.Connection do
-  @moduledoc false
+  @moduledoc """
+  Neuron.Connection is a behaviour for defining pluggable connectors for Neuron library.
+  """
 
-  def post(nil, _, _) do
-    raise ArgumentError, message: "you need to supply an url"
-  end
-
-  def post(url, query, %{headers: headers, connection_opts: connection_opts}) do
-    HTTPoison.post(
-      url,
-      query,
-      headers,
-      connection_opts
-    )
-  end
+  @callback call(String.t(), Keyword.t()) :: {:ok, term} | {:error, term}
```

Comment:
I think we can be a bit more specific here to make sure we return a `Neuron.Response.t()` for successful cases, do you think so?

Reply:
nice idea, changed 👍